lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -513,7 +513,7 @@ sub _generate_jobs {
     }
 
     if (!@products) {
-        carp "no products found for " . join('-', map { $args->{$_} } qw(DISTRI VERSION FLAVOR ARCH));
+        push(@$notes, "no products found for " . join('-', map { $args->{$_} } qw(DISTRI FLAVOR ARCH)));
```

Comment:
Wouldn't it make more sense to return this as error message like `return {error_message => …}`. I suppose the same counts for the `carp` further down.